src/util/taproot.rs

```diff
@@ -772,23 +776,42 @@ impl LeafVersion {
     // or an opcode that is not valid as the first opcode).
     // The values that comply to this rule are the 32 even values between
     // 0xc0 and 0xfe and also 0x66, 0x7e, 0x80, 0x84, 0x96, 0x98, 0xba, 0xbc, 0xbe
-    pub fn from_u8(ver: u8) -> Result<Self, TaprootError> {
-        if ver & TAPROOT_LEAF_MASK == ver && ver != 0x50 {
-            Ok(LeafVersion(ver))
-        } else {
-            Err(TaprootError::InvalidTaprootLeafVersion(ver))
+    pub fn from_byte(version: u8) -> Result<Self, TaprootError> {
+        match version {
+            TAPROOT_LEAF_TAPSCRIPT => Ok(LeafVersion::TapScript),
+            TAPROOT_ANNEX_PREFIX => Err(TaprootError::InvalidTaprootLeafVersion(TAPROOT_ANNEX_PREFIX)),
+            even if even & TAPROOT_LEAF_MASK != even => Err(TaprootError::InvalidTaprootLeafVersion(even)),
```

Comment:
nit: perhaps you want to name this `odd`?

Reply:
Note that the rustdoc says 'even' as well.

Reply:
Nice catch, the doc-comment should have said `odd`

Reply:
Also the parameter name needs to be updated in the docs.

Reply:
@dr-orlovsky, can you please address this. I am good with the rest of the PR. 
- Update the docs here (change to odd instead of even)
- The code variable name should be odd instead of even

Reply:
Thank you for spotting! Addressed with the new force-pushed version.